python-package/lightgbm/engine.py

```diff
@@ -341,7 +341,9 @@ def train(
                 )
         except callback.EarlyStopException as earlyStopException:
             booster.best_iteration = earlyStopException.best_iteration + 1
-            evaluation_result_list = earlyStopException.best_score
+            # eval results from cv() have a 5th element with the standard deviation of metrics,
+            # which is not needed for early stopping
+            evaluation_result_list = [item[:4] for item in earlyStopException.best_score]
```

Comment:
I'm OK with this modification that bring some overhead because this line is executed only once after early stopping had occurred.

Reply:
Yeah thank you, I agree. This will also hopefully be able to go away completely as part of #6748, like I mentioned in https://github.com/microsoft/LightGBM/pull/6900#discussion_r2064047977